yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -454,10 +478,16 @@ private[spark] class Client(
      *   (3) whether to add these resources to the classpath
      */
     val cachedSecondaryJarLinks = ListBuffer.empty[String]
+    val files = sparkConf.get(FILES_TO_DISTRIBUTE).map(p => Utils.resolveURIs(p))
+      .orElse(sys.env.get("SPARK_YARN_DIST_FILES"))
+      .orNull
+    val archives = sparkConf.get(ARCHIVES_TO_DISTRIBUTE).map(p => Utils.resolveURIs(p))
+      .orElse(sys.env.get("SPARK_YARN_DIST_ARCHIVES"))
+      .orNull
     List(
-      (args.addJars, LocalResourceType.FILE, true),
-      (args.files, LocalResourceType.FILE, false),
-      (args.archives, LocalResourceType.ARCHIVE, false)
+      (sparkConf.getOption("spark.jars").orNull, LocalResourceType.FILE, true),
```

Comment:
@tgravescs @vanzin I think maybe I already put "spark.jars" into distributed cached IIUC. Maybe the problem is that for yarn cluster mode, here I both do the distributed cache and SparkContext#addJars things.